lib/heap_sort.rb

```diff
@@ -1,8 +1,18 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity: O(n * log m)?
+# Space Complexity: O(n) ?
+def heapsort(list)
```

Comment:
What is `m` when you talk about time complexity?  Since at worst you have approximately `n` elements in the heap the time complexity is O(n log n).  

Otherwise well done!